### PR TITLE
New component: Checkbox

### PR DESCRIPTION
**Pull request by RodriP**
## Motivation
Making more componentes available! 🎉
## Description
Checkbox is implemented for Simple variation. All details can be found in the RFC at #80
## Affected component
This is a new component.
## Frontify component link
Link to the frontify is [here](https://company-161429.frontify.com/d/kxHCRixezmfK/n-a#/componentes/checkbox)
## Screenshots
ObjC:
image

Swift:
image

## Checks
Are you sure you followed all those steps? If so, repeat after me "I solemnly swear that ...":
   - [x] I have coded an example inside the Demo App,
   - [x] I have added proper tests,
   - [ ] I have modified the Changelog.md file,
   - [ ] I have updated GitHub wiki,
   - [ ] I have the explicit approval of one or more members of the UX Team,
   - [x] I have checked that this code is ObjC compliant.
   - [x] I have checked that all the new public enums conform to AndesEnumStringConvertible.
## Change type
This is easy, let us know what this code is about:
   - [x] This code adds a new component
   - [ ] This code includes improvements to an existent component
   - [ ] This code improves or modifies ONLY the Demo App.

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: LibraryComponents/Classes/AndesCheckbox/Status/AndesCheckboxStatus.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//
+//  AndesCheckboxStatus.swift
+//  AndesUI
+//
+//  Created by Rodrigo Pintos Costa on 6/16/20.
+//
+
+import Foundation
+//Used to set the checkbox status
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/Status/AndesCheckboxStatus.swift:9]:
```suggestion
/// Used to set the checkbox status
```

File: LibraryComponents/Classes/AndesCheckbox/Type/AndesCheckboxType.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
+//
+//  AndesCheckboxType.swift
+//  AndesUI
+//
+//  Created by Rodrigo Pintos Costa on 6/16/20.
+//
+
+import Foundation
+//Used to set the checkbox type
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/Type/AndesCheckboxType.swift:9]:
```suggestion
/// Used to set the checkbox type
```

File: LibraryComponents/Classes/AndesCheckbox/Type/AndesCheckboxTypeDisabled.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
+//
+//  AndesCheckboxDisabledType.swift
+//  AndesUI
+//
+//  Created by Rodrigo Pintos Costa on 6/16/20.
+//
+
+import Foundation
+//Handle checkbox disabled type
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/Type/AndesCheckboxTypeDisabled.swift:9]:
```suggestion
/// Handle checkbox disabled type
```

File: LibraryComponents/Classes/AndesCheckbox/Type/AndesCheckboxTypeError.swift [hunk 4]
```diff
@@ -0,0 +1,17 @@
+//
+//  AndesCheckboxTypeError.swift
+//  AndesUI
+//
+//  Created by Rodrigo Pintos Costa on 6/16/20.
+//
+
+import Foundation
+// Handle checkbox error type
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/Type/AndesCheckboxTypeError.swift:9]:
```suggestion
/// Handle checkbox error type
```

File: LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+class AndesCheckboxDefaultView: UIView, AndesCheckboxView {
+    @IBOutlet weak var label: UILabel!
+
+    @IBOutlet weak var rightButton: UIButton!
+
+    @IBOutlet weak var leftButton: UIButton!
+
+    @IBOutlet var checkboxView: UIView!
+    internal var backgroundLayer: CALayer
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift:18]: Esta property la veo innecesaria, solo la estas usando para setear el `cornerRadius` de un botón u otro, pero eso lo podes hacer directamente al crear la vista, seteando el `cornerRadius` de ambos botones

File: LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+            self.rightButton.isHidden = true
+            self.leftButton.isHidden = false
+            self.leftButton.backgroundColor = config.backgroundColor
+            self.backgroundLayer = self.leftButton.layer
+        } else {
+            self.rightButton.isHidden = false
+            self.leftButton.isHidden = true
+            self.rightButton.backgroundColor = config.backgroundColor
+            self.backgroundLayer = self.rightButton.layer
+
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift:93]:
```suggestion
```

File: LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+        self.delegate?.checkboxTapped()
+    }
+
+    @IBAction func rightButtonTapped(_ sender: Any) {
+        self.delegate?.checkboxTapped()
+    }
+
+    func loadNib() {
+           let bundle = AndesBundle.bundle()
+           bundle.loadNibNamed("AndesCheckboxDefaultView", owner: self, options: nil)
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift:44]: Quedó con doble tab

File: LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+        checkboxView.pinToSuperview()
+        checkboxView.translatesAutoresizingMaskIntoConstraints = false
+        updateView()
+    }
+
+    func updateView() {
+        clearView()
+        self.label.text = config.title
+        let checkboxIcon: String? = config.icon
+        let iconColor: UIColor? = config.iconColor
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift:81]: Con el `if let` que tenes más abajo no es necesario declarar estas dos variables

File: LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+            self.backgroundLayer = self.leftButton.layer
+        } else {
+            self.rightButton.isHidden = false
+            self.leftButton.isHidden = true
+            self.rightButton.backgroundColor = config.backgroundColor
+            self.backgroundLayer = self.rightButton.layer
+
+        }
+
+        if let currentIcon = checkboxIcon, !currentIcon.isEmpty {
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift:96]: `if let currentIcon = config.icon, !currentIcon.isEmpty`

File: LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+            if let currentIconColor = iconColor {
+                if self.leftButton.isHidden == false {
+                    AndesIconsProvider.loadIcon(name: currentIcon) {
+                        checkboxIcon in self.leftButton.setImage(checkboxIcon, for: .normal)
+                    }
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift:101]:
```suggestion
                    AndesIconsProvider.loadIcon(name: currentIcon) { checkboxIcon in
                        self.leftButton.setImage(checkboxIcon, for: .normal)
                    }
```

File: LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+                    self.leftButton.tintColor = currentIconColor
+                } else {
+                    AndesIconsProvider.loadIcon(name: currentIcon) {
+                        checkboxIcon in self.rightButton.setImage(checkboxIcon, for: .normal)
+                    }
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift:106]:
```suggestion
                    AndesIconsProvider.loadIcon(name: currentIcon) { checkboxIcon in 
                        self.rightButton.setImage(checkboxIcon, for: .normal)
                    }
```

File: LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+                    self.rightButton.tintColor = currentIconColor
+                }
+            }
+        }
+        self.label.setAndesStyle(style: AndesStyleSheetManager.styleSheet.bodyM(color: config.textColor))
+        self.backgroundLayer.cornerRadius = config.cornerRadius
+        let borderColor: UIColor? = config.borderColor
+        let borderSize: CGFloat? = config.borderSize
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift:114]: Estas dos variables se pueden borrar también, declarándolas directamente en el `if let`

File: LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+        if let currentBorderColor = borderColor, let currentBorderSize = borderSize {
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift:115]:
```suggestion
        if let currentBorderColor = config.borderColor, let currentBorderSize = config.borderSize {
```

File: LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+            if self.leftButton.isHidden == false {
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesCheckbox/View/AndesCheckboxDefaultView.swift:116]:
```suggestion
            if !self.leftButton.isHidden {
```

**Comment by samuelsainz** [comment]
@nrostanml podes revisar mis últimos dos commits? si estás ok ya mergeamos

**Pull request merged**